cmd/system-probe/api/module/loader_linux.go

```diff
@@ -22,15 +22,31 @@ func isEBPFRequired(factories []Factory) bool {
 	return false
 }
 
+func isEBPFOptional(factories []Factory) bool {
+	for _, f := range factories {
+		if f.OptionalEBPF {
+			return true
+		}
+	}
+	return false
+}
+
 func preRegister(_ *sysconfigtypes.Config, moduleFactories []Factory) error {
-	if isEBPFRequired(moduleFactories) {
-		return ebpf.Setup(ebpf.NewConfig())
+	needed := isEBPFRequired(moduleFactories)
+	if needed || isEBPFOptional(moduleFactories) {
+		err := ebpf.Setup(ebpf.NewConfig())
+		if err != nil && !needed {
```

Comment:
I'm not sure this is sufficient. If ebpf is optional and setup fails, doesn't that need to be communicated to the module so that it doesn't attempt to use ebpf?